src/middlewared/middlewared/etc_files/local/smb4.conf

```diff
@@ -0,0 +1,330 @@
+#
+# SMB.CONF(5)		The configuration file for the Samba suite 
+# $FreeBSD$
+#
+<%
+        import os
+        import sys
+        sys.path.extend([
+            '/usr/local/www',
+            '/usr/local/www/freenasUI'
+        ])
+        import logging
+        logger = logging.getLogger(__name__)
+
+        from freenasUI.common.freenassysctl import freenas_sysctl as fs
+        parsed_conf = {}
+
+        def get_db_config():
+            conf = {}
+
+            conf['cifs'] = middleware.call_sync('datastore.query', 'services.cifs', None, {'get': True})
+            conf['ad'] = middleware.call_sync('datastore.query', 'directoryservice.activedirectory', None, {'get': True})
+            conf['ldap'] = middleware.call_sync('datastore.query', 'directoryservice.ldap', None, {'get': True})
```

Comment:
In case you ever want a short for this:

`middleware.call_sync('datastore.config', 'directoryservice.ldap')`

 would mean the same.